src/mistral_common/tokens/tokenizers/instruct.py

```diff
@@ -646,6 +667,69 @@ def encode_assistant_message(
         """
         return super().encode_assistant_message(message, False, continue_message)
 
+    @overload
+    def _encode_content_chunk(
+        self, chunk: Union[str, TextChunk, ThinkChunk], rstrip: bool
+    ) -> Tuple[List[int], None]: ...
+    @overload
+    def _encode_content_chunk(
+        self, chunk: Union[ImageChunk, ImageURLChunk], rstrip: bool
+    ) -> Tuple[List[int], np.ndarray]: ...
+    @overload
+    def _encode_content_chunk(
+        self, chunk: Union[AudioChunk, AudioURLChunk], rstrip: bool
+    ) -> Tuple[List[int], Audio]: ...
+    def _encode_content_chunk(
+        self, chunk: Union[str, ContentChunk], rstrip: bool
+    ) -> Union[Tuple[List[int], None], Tuple[List[int], np.ndarray], Tuple[List[int], Audio]]:
+        if isinstance(chunk, str):
+            text = chunk.rstrip(" ") if rstrip else chunk
+            return self.tokenizer.encode(text, bos=False, eos=False), None
+        elif isinstance(chunk, TextChunk):
+            text = chunk.text.rstrip(" ") if rstrip else chunk.text
+            return self.tokenizer.encode(text, bos=False, eos=False), None
+        elif isinstance(chunk, ThinkChunk):
+            return self.encode_think(chunk), None
+        elif isinstance(chunk, (ImageChunk, ImageURLChunk)):
+            assert self.image_encoder is not None, "Make sure to define a image encoder at init"
+            img_encoding = self.image_encoder(chunk)
+
+            return img_encoding.tokens, img_encoding.image
+
+        elif isinstance(chunk, (AudioChunk, AudioURLChunk)):
+            # the following is only possible for >= v7
+            assert self.audio_encoder is not None, "Make sure to define a audio encoder at init"
+            audio_encoding = self.audio_encoder(chunk)
+
+            return audio_encoding.tokens, audio_encoding.audio
+        else:
+            raise ValueError(f"Unknown chunk type: {chunk}")
+
+    def _encode_content_chunks(
+        self, content: Sequence[ContentChunk], allowed_types: Optional[Tuple[Type[ContentChunk], ...]], rstrip: bool
+    ) -> Tuple[List[int], List[np.ndarray], List[Audio]]:
+        tokens: List[int] = []
+        images: List[np.ndarray] = []
+        audio: List[Audio] = []
+
+        for chunk in content:
+            if allowed_types and not isinstance(chunk, allowed_types):
+                raise ValueError(f"Invalid chunk type: {chunk}")
```

Comment:
```suggestion
            if allowed_types and not isinstance(chunk, allowed_types):
                raise ValueError(f"Invalid chunk type: {chunk}")
```

ok! Guess we this should normally be in a validator but that works for now!

Reply:
Turns out it was highly unnecessary with all the changes done